File: examples/ios-example/ios-example/ios-example/AuthProvider.swift
import Foundation
import firestore_kmp

class AuthProvider: FirestoreAuth {
    // TODO: Implement token retrieval from Firebase Auth
    // For example, using Firebase iOS SDK:
    // func getIdToken() async throws -> String? {
    //     return try await Auth.auth().currentUser?.getIDToken()
    // }
    
    func getIdToken() async throws -> String? {
        return FirestoreConfig.serviceAccountToken
    }
    
    func getFreshIdToken() async throws -> String? {
        return try await getIdToken()
    }
}

File: examples/android-example/app/src/main/java/com/firestore/kmp/example/AuthProvider.kt
package com.firestore.kmp.example

import com.firestore.kmp.auth.FirestoreAuth

class AuthProvider : FirestoreAuth {
    override suspend fun getIdToken(): String? {
        return FirestoreConfig.SERVICE_ACCOUNT_TOKEN
    }
    
    override suspend fun getFreshIdToken(): String? {
        return FirestoreConfig.SERVICE_ACCOUNT_TOKEN
    }
}

